portal/models/client.py

```diff
@@ -0,0 +1,251 @@
+import base64
+import hashlib
+import hmac
+import json
+import time
+from flask import current_app, abort
+from urlparse import urlparse
+
+from .auth import Token
+from ..database import db
+from ..extensions import oauth
+from ..factories.celery import create_celery
+from .intervention import Intervention, UserIntervention
+from .relationship import RELATIONSHIP
+
+
+class Client(db.Model):
+    __tablename__ = 'clients'  # Override default 'client'
+    client_id = db.Column(db.String(40), primary_key=True)
+    client_secret = db.Column(db.String(55), nullable=False)
+
+    user_id = db.Column(db.ForeignKey('users.id'), nullable=False)
+    user = db.relationship('User')
+
+    _redirect_uris = db.Column(db.Text)
```

Comment:
Might be better to make this a proper table. IIRC [we're pinning an old version of `validators` module](https://github.com/uwcirg/true_nth_usa_portal/blob/develop/requirements.txt#L97) because WTForms can only validate a "proper" field (not `property`)

...probably not worth the effort though :)